js/src/components/elements/select.js

```diff
@@ -16,23 +16,21 @@ export default class Select extends Component {
   constructor({ onChange }) {
     super();
     // @fixme State should be stored in this.state
-    this.selected = new Set();
     this.onChange = onChange;
   }
   changeHandler = (e) => {
-    if (this.selected.has(e.target.value)) {
-      this.selected.delete(e.target.value);
-    }
-    else {
-      this.selected.add(e.target.value);
-    }
-    this.onChange(this.selected);
+    const selected = new Set(
+      Array.from(e.target.options)
+        .filter(option => option.selected)
+        .map(option => option.value)
+    );
+    this.onChange(selected);
   }
   render() {
-    const { label, data } = this.props;
+    const { label, data, selected } = this.props;
```

Comment:
So I'm not a huge fan of this pattern. If you're not careful you can miss this deconstruct and then wonder where `label`, `data`, and `selected` are coming from.